subsys/modbus/modbus_serial.c

```diff
@@ -337,6 +337,17 @@ static void cb_handler_rx(struct modbus_context *ctx)
 		}
 
 	} else {
+		if (cfg->uart_buf_ctr == CONFIG_MODBUS_BUFFER_SIZE) {
+			/*
+			 * Input buffer is full, but more data is arriving.
+			 * Read and discard a byte from the fifo to clear the RX
+			 * interrupt. Leave the timer running to process the
+			 * buffer.
+			 */
+			uint8_t c;
+			(void)uart_fifo_read(cfg->dev, &c, 1);
+			return;
```

Comment:
I think if we get flooded here, then RX interrupts should be disabled instead of just dropping characters.